databricks-sdk-java/src/main/java/com/databricks/sdk/core/commons/CommonsHttpClient.java

```diff
@@ -117,18 +122,26 @@ private HttpUriRequest transformRequest(Request in) {
       case Request.DELETE:
         return new HttpDelete(in.getUri());
       case Request.POST:
-        return withEntity(new HttpPost(in.getUri()), in.getBody());
+        return withEntity(new HttpPost(in.getUri()), in);
       case Request.PUT:
-        return withEntity(new HttpPut(in.getUri()), in.getBody());
+        return withEntity(new HttpPut(in.getUri()), in);
       case Request.PATCH:
-        return withEntity(new HttpPatch(in.getUri()), in.getBody());
+        return withEntity(new HttpPatch(in.getUri()), in);
       default:
         throw new IllegalArgumentException("Unknown method: " + in.getMethod());
     }
   }
 
-  private HttpRequestBase withEntity(HttpEntityEnclosingRequestBase request, InputStream body) {
-    request.setEntity(new InputStreamEntity(body));
+  private HttpRequestBase withEntity(HttpEntityEnclosingRequestBase request, Request in) {
+    if (in.isBodyString()) {
+      request.setEntity(new StringEntity(in.getBodyString(), StandardCharsets.UTF_8));
+    } else if (in.isBodyStreaming()) {
+      request.setEntity(new InputStreamEntity(in.getBodyStream()));
+    } else {
+      LOG.warn(
+          "withEntity called with a request with no body, so no request entity will be set. URI: {}",
```

Comment:
Right now, `withEntity` is only being called in POST / PUT and PATCH right? Is this pathway only for the PUT?

Reply:
`withEntity` should be called for any HTTP methods that have a request body, i.e. POST/PUT/PATCH. GET/DELETE and other methods we may support in the future like HEAD/OPTIONS don't support request bodies, so we don't use this method.